rfcs/text/0049-entity-fields.md

```diff
@@ -20,31 +21,55 @@ This approach would allow ECS to accommodate new types of entities without requi
 
 | Field | Type | Description |
 |-------|------|-------------|
+| entity.definition_id | keyword | Used Elastic solutions (e.g., Security, Observability) to denote the ID of the entity definition which is used to extract entity details from ingested logs, events, intelligence, and other data types. Use of this value is reserved, and ECS producers, including data ingestion pipelines, must not populate this field|
```

Comment:
I'm not too sure about this field being reserved. There are no other fields in ECS that are reserved for use by Elastic, and I don't really know if it makes sense to have them. Since the intention of ECS is a common schema that's shared with others, I don't if this would make sense to include.

Are there alternatives to adding this, such as using a custom field that's not defined in ECS?

Reply:
Yes, I agree, it's an internal-only field - no need to have it defined in the ECS spec.  We already intend to publish the entity store schema in our docs.  Any problem with using `entity.Definition_id` as a custom field? (nesting a custom leaf field under an ECS-defined root object `entity.*`